python/rmm/rmm/pylibrmm/stream.pyi

```diff
@@ -0,0 +1,16 @@
+# SPDX-FileCopyrightText: Copyright (c) 2020-2025, NVIDIA CORPORATION.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Optional
+
+class Stream:
+    def __init__(self, obj: Optional[Any] = None) -> None: ...
```

Comment:
I didn't attempt to model `obj` correctly. I think the technically allowed things are

1. Other streams
2. numba streams
3. cupy streams
4. objects implementing the `__cuda_stream__` protocol

Once everything implements `__cuda_stream__` (which we already do) I think we can make this type more  meaningful.